Changes to docs due to 0.20/1 deprecation (#2534)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/core-versions.md

## Version support prior to v1.0

All dbt Core versions released prior to 1.0 and their version-specific documentation have been deprecated. If upgrading to a currently supported version, reference our [best practices for upgrading](#best-practices-for-upgrading)

## Current version support

### Minor versions

> Minor versions include new features and capabilities. They will be supported for one year from the date of their initial release. _dbt Labs is committed to this 12 month support window._ Our mechanism for continuing to support a minor version is by releasing new patches: small, targeted bug fixes. Whenever we refer to a minor version, such as v1.0, we always mean its latest available patch release (v1.0.x).

Author: ```suggestion
Minor versions include new features and capabilities. They will be supported for one year from their initial release date. _dbt Labs is committed to this 12-month support timeframe._ Our mechanism for continuing to support a minor version is by releasing new patches: small, targeted bug fixes. Whenever we refer to a minor version, such as v1.0, we always mean its latest available patch release (v1.0.x).
```

While a minor version is officially supported:
- You can use it in dbt Cloud. For more on dbt Cloud versioning, see [Choosing a dbt version](cloud-choosing-a-dbt-version).
- You can select it from the version dropdown on this website, to see documentation that is accurate for use with that minor version.

### Ongoing patches

> During the 12 month support window, we will continue to release new patch versions that include fixes.

Author: ```suggestion
During the 12-month support window, we will continue to release new patch versions that include fixes.
```

### Upgrading to new minor versions

> During the official support window period, minor versions will remain available in dbt Cloud, and in the version dropdown on the docs site. While we do not expect users to immediately upgrade to newer minor versions as soon as they're available, there will always be some features and fixes that are only available for users of the latest minor version.

Author: ```suggestion
During the official support period, minor versions will remain available in dbt Cloud and the version dropdown on the docs site. While we do not expect users to immediately upgrade to newer minor versions as soon as they're available, there will always be some features and fixes only available for users of the latest minor version.
```

Like many software projects, dbt Core releases follow [semantic versioning](https://semver.org/), which defines three types of version releases.

> - **Major versions:** To date, dbt Core has had one major version release: v1.0.0. When v2.0.0 is released, it will introduce new features and backwards compatibility for functionality that has been deprecated will no longer work.

Reviewer: Hm, "backwards compatibility ... will no longer work" doesn't sound quite right to my ear — maybe something like:
```suggestion
- **Major versions:** To date, dbt Core has had one major version release: v1.0.0. When v2.0.0 is released, it will introduce new features, and functionality that has been announced for deprecation will stop working.
```

### How we version adapter plugins

When you use dbt, you use a combination of `dbt-core` and an adapter plugin specific to your database. You can see the current list in [Supported Data Platforms](supported-data-platforms). Both `dbt-core` and dbt adapter plugins follow semantic versioning.

`dbt-core` and adapter plugins coordinate new features and behind-the-scenes changes in minor releases. When fixing bugs, sooner is better, so patch versions are released independently for `dbt-core` and plugins.

That means that patch version numbers will likely differ between `dbt-core` and the adapter plugin(s) you have installed. However, major and minor version numbers should always match.

> For example, you may find you're using `dbt-core==1.2.3` with `dbt-snowflake==1.2.0`. It is critical that you're using the latest patch available for both core and the adapter (v1.2.x). Use the `dbt --version` command to see which versions you have installed:

Reviewer: Good point. How should we keep these numbers up-to-date? :) I guess we could add to the release checklist for new minor versions

Author: Will add it to the list!
